src/instructlab/utils.py

```diff
@@ -900,3 +914,73 @@ def list_models(
 def contains_argument(prefix: str, args: typing.Iterable[str]) -> bool:
     # Either --foo value or --foo=value
     return any(s == prefix or s.startswith(prefix + "=") for s in args)
+
+
+def get_model_arch(model_path: pathlib.Path) -> str:
+    """
+    Extract a given model's architecture from its config if available and return it
+
+    args
+        model_path (str): Path to the model, used to read the model config if available
+    returns
+        model_arch (str): The architecture of the model as expressed in the config file
+    """
+
+    # TODO: implement for GGUF models
+
+    model_arch = "default"
+    try:
+        model_cfg = get_config_file_from_model(model_path, "config.json")
+        model_arch = model_cfg["model_type"]
+    except Exception as e:
+        logger.debug(
+            f"Unable to get model architecture from config for model: {model_path}: {e}. Falling back to default value"
+        )
+
+    return model_arch
+
+
+def get_config_file_from_model(model_path: pathlib.Path, filename: str):
+    try:
+        with open(
+            pathlib.Path(model_path) / filename,
+            "r",
+            encoding="utf-8",
+        ) as f:
+            model_cfg = json.load(f)
+        return model_cfg
+    except (FileNotFoundError, NotADirectoryError, PermissionError) as e:
+        raise ValueError(
+            f"Unable to load file {filename} from {model_path}: {e}"
+        ) from e
+
+
+def get_model_template_from_tokenizer(model_path: pathlib.Path) -> Tuple[str, str, str]:
+    template = eos_token = bos_token = None
+
+    try:
+        tcfg = get_config_file_from_model(model_path, "tokenizer_config.json")
+        template = tcfg["template"]
+        bos_token = tcfg["bos_token"]
+        eos_token = tcfg["eos_token"]
+    except (FileNotFoundError, NotADirectoryError, PermissionError) as e:
+        raise e
+
+    return template, eos_token, bos_token
+
+
+def use_legacy_pretraining_format(model_path: Path, model_arch: str) -> bool:
```

Comment:
we should scope getting rid of all of this and linux_train in general

Reply:
linux train yes
this function has been specifically added to handle an edge case for as long as we need to keep compatibility with granite-7b models